dist/browser.js

```diff
@@ -76,177 +76,184 @@ var util = require('../../util');
+                            iframe.parentNode.style.marginBottom = '30px';
+                        } else {
+                            // Html will be empty if store has run out of free sales credits.
+                            // Gracefully fail to load widget by removing the iframe from DOM.
+                            iframe.parentNode.removeChild(iframe);
+                        }
+                    } else {
+                        console.log('Error: ' + statusText);
+                        // Remove iframe from DOM
+                        iframe.parentNode.removeChild(iframe);
+                    }
+                });
+        }.bind(this));
+    }
 
-  customIntegration() {
-    if (this.tokenId) {
-      if (this.product) {
-        // Update product
-        // $.get(this.baseUrl+this.productEndpoint)
-        // .done(function(data) {
-        //     console.log(data);
-        // });
-      }
-      else if (this.order) {
-        var token = this.tokenId;
-        // Submit order
-        $.ajaxSetup({
-          beforeSend: function(xhr, settings) {
-            xhr.setRequestHeader("X-CSRFToken", token);
-          }
-        });
-        $.ajax({
-          type: "POST",
-          url: this.baseUrl + this.orderEndpoint,
-          data: JSON.stringify(this.order),
-          contentType: "application/json; charset=utf-8",
-          dataType: "json",
-          success: function (msg) {
-            console.log(msg);
-          },
-          error: function (errormessage) {
-           console.log(errormessage);
-          }
-        });
-      }
+    customIntegration() {
+        if (this.tokenId) {
+            var url, data;
+            if (this.product) {
+                // Update product
+                url = this.opts.baseUrl + this.opts.productEndpoint;
+                data = this.product;
+            } else if (this.order) {
+                // Submit order
+                url = this.opts.baseUrl + this.opts.orderEndpoint;
+                data = this.order;
+            }
+            $.ajax({
+                type: "POST",
+                url: url,
+                data: JSON.stringify(data),
+                beforeSend: function(xhr, settings) {
+                    xhr.setRequestHeader("X-CSRFToken", this.tokenId);
+                    xhr.setRequestHeader('Content-Type', 'application/json; charset=utf-8');
+                },
+                dataType: "json",
+                statusCode: {
+                    200: function() {
+                        console.log("Success");
+                    },
+                    400: function() {
+                        console.log("Missing required fields");
+                    },
+                    500: function() {
+                        console.log("Internal server error");
+                    },
+                },
+                success: function (data, textStatus, jqXHR) {
+                    console.log(textStatus);
+                },
```

Comment:
Seems to not be handling error completely here. `success` callback is good, but missing `error` callback.

For `statusCode` the functions get the same params as success and error `(data, textStatus, jqXHR)`. I think `error()` will handle 400 codes, and `success()` will handle 200 codes. So I think we only need `statusCodes.500`